pyro/poutine/broadcast_messenger.py

```diff
@@ -63,4 +63,6 @@ def _pyro_sample(msg):
             for i in range(-len(target_batch_shape) + 1, 1):
                 if target_batch_shape[i] is None:
                     target_batch_shape[i] = actual_batch_shape[i] if len(actual_batch_shape) >= -i else 1
-            msg["fn"] = msg["fn"].expand(target_batch_shape)
+            msg["fn"] = dist.expand(target_batch_shape)
+            if "has_rsample" in dist.__dict__:
```

Comment:
@neerajprad This is kind of gross, but otherwise we'd need to update O(#distributions)-many implementations of `.expand()`. Any ideas for alternatives?

Reply:
I think we can probably add this check to `TorchDistribution._get_checked_instance`, but we can do it in PyTorch directly to make it available for PyTorch users.

Reply:
@neerajprad that's a great idea. I've made the logic a little safer, but I do like the idea of updating `._get_checked_instance` to copy this and possibly other custom attributes; that way other possible occurrences of `.expand()` won't silently break.

Reply:
@neerajprad on further inspection, `_get_checked_instance()` doesn't seem like quite the right place. Rather I think we can implement a revised interface that copies `._validate_args` and any other of a set of whitelisted attributes. Of course this is all PyTorch work, out of scope for this PR.